py/plugins/google-genai/src/genkit/plugins/google_genai/models/gemini.py

```diff
@@ -528,6 +532,38 @@ def _call_tool(self, call: genai_types.FunctionCall) -> genai_types.Content:
             ]
         )
 
+    async def _retrieve_cached_content(
+        self, request: GenerateRequest, model_name: str, cache_config: dict, contents: list[genai_types.Content]
+    ) -> genai_types.CachedContent | None:
+        validate_context_cache_request(request=request, model_name=model_name)
+
+        ttl = cache_config.get('ttl_seconds', DEFAULT_TTL)
+        cache_key = generate_cache_key(request=request)
+
+        iterator_config = genai_types.ListCachedContentsConfig()
+        cache = None
+        pages = await self._client.aio.caches.list(config=iterator_config)
+
+        async for item in pages:
```

Comment:
is there a way to look up the cache by name? I don't remember exactly why in JS we also list... would be nice if we didn't have to do it.

Reply:
Yeah, so the problem is that we cannot force to set particular `name` for the context caching instance. Instead, I am setting `display_name` and then have to iterate over cache to find matching `display_name` and this way to get the actual name.

In JS its similar, perhaps once we did this operation in our runtime we can do some caching but no native option afaik